caffe2/proto/torch.proto

```diff
@@ -4,547 +4,91 @@ import "caffe2/proto/caffe2.proto";
-  // Submodules and methods in PyTorch.
-  repeated GraphProto methods = 15;
-
-  // Named metadata values; keys should be distinct.
-  // Many meta data in MetaNetDef and preditor are piggy backed here.
-  // 1) project
-  // 2) model_class
-  // 3) internal_version
-  // 4) predictor_type
-  // 5) predictor_id
-  // 6) execute_plan
-  // 7) applicationSpecificInfo (another string map, need to verify it has no duplicate.)
-  // 8) engine
-  // 9) publish time
-  repeated StringStringEntryProto metadata_props = 14;
-
-  // Model name
-  optional string name = 16;
-
-  // Model name
-  repeated AttributeProto annotations = 17;
-
-  // Mapping from list name to blob name list, must be string list type.
-  // Equivalent to blobs in MetaNetDef.
-  repeated AttributeProto blob_lists = 51;
-
-  // Mapping from plan name to serialized plan, must be string list type.
-  // Equivalent to plans in MetaNetDef.
-  repeated AttributeProto plans = 52;
-};
-
-// StringStringEntryProto follows the pattern for cross-proto-version maps.
-// See https://developers.google.com/protocol-buffers/docs/proto3#maps
-message StringStringEntryProto {
-  optional string key = 1;
-  optional string value= 2;
-};
-
-// Graphs
-//
-// A graph defines the computational logic of a model and is comprised of a parameterized
-// list of nodes that form a directed acyclic graph based on their inputs and outputs.
-// This is the equivalent of the "network" or "graph" in many deep learning
-// frameworks.
-// Graph ==> NetDef in Caffe2
-//       ==> Submodule/Method in PyTorch
-message GraphProto {
-  // The nodes in the graph, sorted topologically.
-  repeated NodeProto node = 1;
-
-  // The name of the graph.
-  optional string name = 2;   // namespace Graph
-
-  // A list of named tensor values, used to specify constant inputs of the graph.
-  // Each TensorProto entry must have a distinct name (within the list) that
-  // also appears in the input list.
-  repeated TensorProto initializer = 5;
-
-  // A human-readable documentation for this graph. Markdown is allowed.
-  optional string doc_string = 10;
-
-  // The inputs and outputs of the graph.
-  repeated ValueInfoProto input = 11;
-  repeated ValueInfoProto output = 12;
-
-  // Information for the values in the graph. The ValueInfoProto.name's
-  // must be distinct. It is optional for a value to appear in value_info list.
-  repeated ValueInfoProto value_info = 13;
-
-  // Additional annotations.
-  repeated AttributeProto annotations = 14;
-
-  // DO NOT USE the following fields, they were deprecated from earlier versions.
-  // repeated string input = 3;
-  // repeated string output = 4;
-  // optional int64 ir_version = 6;
-  // optional int64 producer_version = 7;
-  // optional string producer_tag = 8;
-  // optional string domain = 9;
+message ParameterDef {
+  // whether we compute the gradient for the parameter
+  optional bool require_gradient = 1;
+  // if keep_in_opt is true, we keep the this parameter
```

Comment:
I don't get what this comment mean

also where did is_buffer go?

Reply:
I will change the name back.